diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index f980ecd3..f239de1a 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -50,7 +50,8 @@ RUN apk add --no-cache \
 		curl \
 		openssl \
 		tar \
-		xz
+		xz \
+		gdb
 
 # ensure www-data user exists
 RUN set -eux; \
@@ -67,6 +68,7 @@ RUN set -eux; \
 		ca-certificates \
 		curl \
 		xz-utils \
+		gdb \
 	; \
 	rm -rf /var/lib/apt/lists/*
 {{ ) end -}}
@@ -414,7 +416,6 @@ RUN set -eux; \
 	cp -v php.ini-* "$PHP_INI_DIR/"; \
 	\
 	cd /; \
-	docker-php-source delete; \
 	\
 {{ if is_alpine then ( -}}
 	runDeps="$( \
